src/main/resources/assets/computercraft/lua/rom/apis/parallel.lua

```diff
@@ -34,7 +66,7 @@ local function runUntilLimit( _routines, _limit )
                     if coroutine.status( r ) == "dead" then
                         _routines[n] = nil
                         living = living - 1
-                        if living <= _limit then
+                        if living <= _limit or (_bundleData and not _bundleData.run) then
```

Comment:
If we add new coroutines to the list, `parallel.waitForAll` will exit when the original number has finished, rather when all of them have. It's probably worth having a "any/all" flag, and then changing the behaviour based on that instead.